external-crates/move/move-bytecode-verifier/src/type_safety.rs

```diff
@@ -809,17 +823,21 @@ fn verify_instr(
 
         Bytecode::VecPack(idx, num) => {
             let element_type = &verifier.resolver.signature_at(*idx).0[0];
-            for _ in 0..*num {
-                let operand_type = safe_unwrap!(verifier.stack.pop());
-                if element_type != &operand_type {
+            if let Some(num_to_pop) = NonZeroU64::new(*num) {
+                let is_mismatched = verifier
+                    .stack
+                    .pop_eq_n(num_to_pop)
+                    .map(|t| element_type != &t)
+                    .unwrap_or(false);
```

Comment:
I think this is wrong? I think this should be an `unwrap_or(true)`.

If there are only `k < n` types in `AbsStack` then `pop_eq_n` will return an `Err(AbsStackError::ElementNotEqual)`

This will then get unwrapped to `false` which will then not trigger the below if statement.

Reply:
How do we not have a failing test for this...

Reply:
How do we not have a test for this... good catch! great catch!

Reply:
Added a test, good catch! Great catch!